File: core/SyncPlatform/src/Services/Iam/Iam.Application/Helpers/RandomAvatarUrl.cs
namespace Iam.Application.Helpers;

/// <summary>
/// Builds avatar URLs compatible with the Flutter <c>random_avatar</c> package (Multiavatar).
/// The stored seed is passed to <c>RandomAvatar(seed)</c> on the client.
/// </summary>
public static class RandomAvatarUrl
{
    public const string Prefix = "randomavatar:";

    /// <summary>
    /// Creates a deterministic Multiavatar seed URL for a newly registered user.
    /// Email is used as the seed so the avatar stays stable across registration retries.
    /// </summary>
    public static string ForRegistration(string email, string? fullName = null)
    {
        var seed = email.Trim().ToLowerInvariant();
        if (string.IsNullOrEmpty(seed) && !string.IsNullOrWhiteSpace(fullName))
            seed = fullName.Trim();

        return $"{Prefix}{seed}";
    }
}

File: ui/app/lib/shared/utils/random_avatar_url.dart
/// Parses avatar URLs produced by IAM registration (`randomavatar:{seed}`).
/// The seed is consumed by the [random_avatar] Multiavatar wrapper on Flutter.
class RandomAvatarUrl {
  RandomAvatarUrl._();

  static const prefix = 'randomavatar:';

  /// Returns the Multiavatar seed when [avatarUrl] uses the IAM random-avatar scheme.
  static String? extractSeed(String? avatarUrl) {
    final url = avatarUrl?.trim();
    if (url == null || url.isEmpty) return null;
    if (!url.startsWith(prefix)) return null;
    final seed = url.substring(prefix.length);
    return seed.isEmpty ? null : seed;
  }
}
